src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/idam/IdamOpenIdClient.java

```diff
@@ -170,4 +166,14 @@ public String nextUserEmail() {
         @SerializedName("access_token")
         private String accessToken;
     }
+
+
+    public static String generateSidamPassword() {
+        String regex = "^(?=.{10,})(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9]).*$";
+        String password = RandomStringUtils.randomAlphanumeric(10);
+        if (!password.matches(regex)) {
+            password = generateSidamPassword();
+        }
```

Comment:
thats not the point.. code is written such that it can go into infinite loop even if the chance is 1 in 1000 times we have to fix it. Any code which can result into an infinite loop in not good in practice and should be refactored. This will break in mutation tests and fortify both

Reply:
Used Generex library code to get the password through given regex which internally also does same recursive way.
This library has been used in Hmcts already.